crates/rattler_solve/src/resolvo/mod.rs

```diff
@@ -378,6 +388,15 @@ impl<'a> DependencyProvider<SolverMatchSpec<'a>> for CondaDependencyProvider<'a>
 
         Dependencies::Known(dependencies)
     }
+
+    fn should_cancel_with_value(&self) -> Option<Box<dyn std::any::Any>> {
```

Comment:
sorry for being slow - why the return type is not `Option<Box<dyn CancelReason>>`  but` dyn Any`?

Reply:
It's defined like that in the trait (in `resolvo`) so I think we need to honor that contract and return an `Box<Any>`